### PR TITLE
Scripts adjusted for docker provisioning for addons

### PR DESCRIPTION
**Pull request by kenan435**

### CODE REVIEW

File: addons/Dockerfile [hunk 1]
```diff
@@ -2,14 +2,13 @@ FROM debian:jessie
...
-COPY kubectl kubectl
+RUN apt-get update && apt-get install -y wget
+RUN wget https://storage.googleapis.com/kubernetes-release/release/v1.2.0/bin/linux/amd64/kubectl
 
-# copy addon yaml definitions
-COPY heapster/ heapster/
-COPY dns/ dns/
-COPY dashboard/ dashboard/
+RUN chmod +x kubectl
+RUN cp kubectl /usr/local/bin/
```
Comment (cheld) [inline addons/Dockerfile]: why not download to the right location?

wget <file.ext> -P /path/to/folder

File: addons/Dockerfile [hunk 2]
```diff
@@ -2,14 +2,13 @@ FROM debian:jessie
 
 WORKDIR /opt/kube-addons
 
-# copy kubectl binary
-COPY kubectl kubectl
+RUN apt-get update && apt-get install -y wget
```
Comment (cheld) [inline addons/Dockerfile:5]: why not use curl? seems to be included

File: addons/run.sh [hunk 3]
```diff
@@ -10,6 +10,17 @@ echo "Addon creation started"
 
 cd ${WORK_DIR}
 
+ENABLE_CLUSTER_MONITORING=${ENABLE_CLUSTER_MONITORING:-"monasca"}
+ENABLE_NODE_LOGGING=${ENABLE_NODE_LOGGING:-"true"}
+LOGGING_DESTINATION=monasca=${LOGGING_DESTINATION:-"monasca"}
+OPENSTACK_USER_ID=${OPENSTACK_USER_ID:-"ff69eeb3fd3b412a917be02ecc833963"}
```
Comment (cheld) [inline addons/run.sh]: Do these defaults make any sense? Are these static for DevStack? Kind-of samples?

File: fragments/master.sh [hunk 4]
```diff
@@ -211,9 +211,55 @@ start_k8s(){
...
+}
+
+
+set +e
+poll_api_server(){
+
+  url='http://localhost:8080/healthz'
+  response=$(curl --write-out %{http_code} --silent --output /dev/null $url )
+  ok_status=200
+  timeout=1800 # 30 minutes
```
Comment (cheld) [inline fragments/master.sh]: 5 minutes, I would suggest

File: fragments/master.sh [hunk 5]
```diff
@@ -211,9 +211,55 @@ start_k8s(){
 
 }
 
+pull_addons_image(){
```
Comment (cheld) [inline fragments/master.sh]: docker image is pulled implicitly - not needed, I assume

Comment (cheld) [inline fragments/master.sh]: hmm..attempt to paralyze wait for api server and pulling image? However, pull should be fast. Only a few bytes, I assume. Is it worth the extra code?

File: fragments/master.sh [hunk 6]
```diff
@@ -211,9 +211,55 @@ start_k8s(){
...
+  else
+        echo "Api service cannot be reched ..."
+  fi
+}
+set -e
+
 echo "Detecting your OS distro ..."
 detect_lsb
 
+set +e
```
Comment (cheld) [inline fragments/master.sh]: Duplicate?

File: fragments/master.sh [hunk 7]
```diff
@@ -211,9 +211,55 @@ start_k8s(){
...
+pull_addons_image &
```
Comment (cheld) [inline fragments/master.sh]: hmm..why not:

-Start api server
-wait until ready
-deploy add-ons

Makes sense?

File: addons/Dockerfile (hunk 2 above)
Comment (kenan435) [inline addons/Dockerfile:5]: No, curl is not included.

File: addons/Dockerfile (hunk 1 above)
Comment (kenan435) [inline addons/Dockerfile]: agree

File: fragments/master.sh (hunk 5 above)
Comment (kenan435) [inline fragments/master.sh]: Taken out. I wanted to make sure that once the the api server responds with 200 we start deploying addons asap, but I think at that point most of the layers will be available.

**Comment by cheld** [comment]
LGTM

**Pull request merged**